logstash-core/lib/logstash/pipeline.rb

```diff
@@ -69,6 +69,7 @@ def initialize(config_str, settings = {})
     @settings = DEFAULT_SETTINGS.clone
     settings.each {|setting, value| configure(setting, value) }
     @reporter = LogStash::PipelineReporter.new(@logger, self)
+    @allow_env = settings[:allow_env] || false
```

Comment:
I'll reuse that.

as a side note now that I am looking at it. the whole `configure(setting)` thing is difficult to grok